### PR TITLE
Adding tests of examples in the Reference Manual

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #3279

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/DafnyRef/Statements.md [hunk 1]
```diff
@@ -1828,10 +1895,10 @@ possible where there might be information overload otherwise.
 
 But then there may be specific instances where the definition of that opaque function is needed. In that situation, the
 body of the function can be _revealed_ using the reveal statement. Here is an example:
-```dafny
+```dafny <!-- %check-verify Statements.9.expect -->
 function {:opaque} f(i: int): int { i + 1 }
 
-method m(int i) {
```
Comment (atomb) [inline docs/DafnyRef/Statements.md:1834]: Someone was still writing C#. :)

**Review by atomb — APPROVED** [review]
This looks good. I like the way `%save` and `%use` allows code to be spread out over multiple blocks. And including the previously-external Dafny files inline means that they'll be rendered when you view the Markdown files on GitHub, which also seems like a plus.

File: docs/DafnyRef/Statements.md (hunk 1 above)
Comment (davidcok) [reply docs/DafnyRef/Statements.md:1834]: Java actually - 20 years of habit die hard

File: docs/check-examples [hunk 2]
```diff
@@ -48,6 +48,10 @@
 ##                  the output is just a success message
 ##              (2) if <expect> is non-empty. then the exit code is 2 and
 ##                  the actual output matches the content of the file <expect>
+##   %check-run <expect> : runs the command
+##              "dafny run --useBaseNameForFileName $F" and, 
+##              if <expect> is present, then the output matches the
```
[the remark this answers is not in the document]
Comment (davidcok) [reply docs/check-examples:53]: I'll include this some other changes coming in the future.

**Pull request merged**